packages/kbn-language-documentation-popover/src/components/as_flyout/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { getFilteredGroups } from '../../utils/get_filtered_groups';
+import { DocumentationMainContent, DocumentationNavigation } from '../shared';
+import { getESQLDocsSections } from '../../sections';
+import type { LanguageDocumentationSections } from '../../types';
+
+interface DocumentationFlyoutProps {
+  isHelpMenuOpen: boolean;
+  onHelpMenuVisibilityChange: (status: boolean) => void;
+  searchInDescription?: boolean;
+  linkToDocumentation?: string;
+}
+
+function DocumentationFlyout({
+  searchInDescription,
+  linkToDocumentation,
+  isHelpMenuOpen,
+  onHelpMenuVisibilityChange,
+}: DocumentationFlyoutProps) {
+  const [documentationSections, setDocumentationSections] =
+    useState<LanguageDocumentationSections>();
+
+  const { euiTheme } = useEuiTheme();
+  const DEFAULT_WIDTH = euiTheme.base * 34;
+
+  const [selectedSection, setSelectedSection] = useState<string | undefined>();
+  const [searchText, setSearchText] = useState('');
+
+  const scrollTargets = useRef<Record<string, HTMLElement>>({});
+
+  const onNavigationChange = useCallback((selectedOptions) => {
+    setSelectedSection(selectedOptions.length ? selectedOptions[0].label : undefined);
+    if (selectedOptions.length) {
+      const scrollToElement = scrollTargets.current[selectedOptions[0].label];
+      scrollToElement.scrollIntoView();
+    }
+  }, []);
+
+  useEffect(() => {
+    onHelpMenuVisibilityChange(isHelpMenuOpen ?? false);
+  }, [isHelpMenuOpen, onHelpMenuVisibilityChange]);
+
+  useEffect(() => {
+    async function getDocumentation() {
+      const sections = await getESQLDocsSections();
+      setDocumentationSections(sections);
+    }
+    if (!documentationSections) {
+      getDocumentation();
+    }
+  }, [documentationSections]);
+
+  const filteredGroups = useMemo(() => {
+    return getFilteredGroups(searchText, searchInDescription, documentationSections, 1);
+  }, [documentationSections, searchText, searchInDescription]);
+
+  return (
+    <>
+      {isHelpMenuOpen && (
+        <EuiFlyout
+          ownFocus
+          onClose={() => onHelpMenuVisibilityChange(false)}
+          aria-labelledby="esqlInlineDocumentationFlyout"
+          type="push"
+          size={DEFAULT_WIDTH}
+        >
+          <EuiFlyoutHeader hasBorder>
+            <DocumentationNavigation
+              searchText={searchText}
+              setSearchText={setSearchText}
+              onNavigationChange={onNavigationChange}
+              filteredGroups={filteredGroups}
+              linkToDocumentation={linkToDocumentation}
+              selectedSection={selectedSection}
+            />
+          </EuiFlyoutHeader>
+          <EuiFlyoutBody>
+            <DocumentationMainContent
+              searchText={searchText}
+              scrollTargets={scrollTargets}
+              filteredGroups={filteredGroups}
+              sections={documentationSections}
+            />
+          </EuiFlyoutBody>
```

Comment:
Can we fix the search/dropdown to the top and make the rest scrolling? I believe that this could simplify the interaction and searches in the docs

Reply:
This is how the flyout works 🤔 In dont understand the comment

Reply:
IF your comment is about the inline experience, I dont like it personally because making it sticky gives very small space to the scolling area. I think this is why Ryan didnt suggest that. I believe though that the solution in my follow up PR https://github.com/stratoula/kibana/pull/33 will make the experience much much better!

Reply:
The comment is about make the search/dropdown stick to the top and make only the documentation scrollable.
Now if I scroll in the documentation the search/dropdown scroll themself with the rest of the content

Reply:
So it is not about the flyout (as it appears in Discover) but in the inline implementation as it is in the inline editing. Because you made it in this file and got confused.

So as I say above:

> I dont like it personally because making it sticky gives very small space to the scolling area. I think this is why Ryan didnt suggest that. I believe though that the solution in my follow up PR https://github.com/stratoula/kibana/pull/33 will make the experience much much better!

cc @ryankeairns